Library/Formula/wxmac.rb

```diff
@@ -69,6 +71,14 @@ def install
 
     system "./configure", *args
     system "make", "install"
+
+    if build.with? "static"
+      args.delete("--enable-shared")
+      args << "--disable-shared"
+
+      system "./configure", *args
+      system "make", "install"
```

Comment:
You're right, it is kinda weird. 
I'm confident that `configure` has to be called twice, because this is the only option related to making static or shared builds:
`--disable-shared        create static library instead of shared`
So, the build with be either-or, but not both.

So `configure` and `make` need to be called twice, but calling `make install` twice might not be necessary.
I'll find out.